tests/ignite/contrib/metrics/test_roc_auc.py

```diff
@@ -1,117 +1,129 @@
 
-    np.random.seed(1)
-    size = 100
-    np_y_pred = np.random.rand(size, 1)
-    np_y = np.zeros((size,), dtype=np.long)
-    np_y[size // 2 :] = 1
-    np.random.shuffle(np_y)
 
-    np_roc_auc = roc_auc_score(np_y, np_y_pred)
+def test_binary_input_N():
 
-    batch_size = 10
+    roc_auc = ROC_AUC()
 
-    def update_fn(engine, batch):
-        idx = (engine.state.iteration - 1) * batch_size
-        y_true_batch = np_y[idx : idx + batch_size]
-        y_pred_batch = np_y_pred[idx : idx + batch_size]
-        return idx, torch.from_numpy(y_pred_batch), torch.from_numpy(y_true_batch)
+    def _test(y_pred, y, n_iters):
+        roc_auc.reset()
+        roc_auc.update((y_pred, y))
 
-    engine = Engine(update_fn)
+        np_y = y.numpy()
+        np_y_pred = y_pred.numpy()
 
-    roc_auc_metric = ROC_AUC(output_transform=lambda x: (x[1], x[2]))
-    roc_auc_metric.attach(engine, "roc_auc")
+        if n_iters > 1:
+            batch_size = y.shape[0] // n_iters + 1
+            for i in range(n_iters):
+                idx = i * batch_size
+                roc_auc.update((y_pred[idx : idx + batch_size], y[idx : idx + batch_size]))
 
-    data = list(range(size // batch_size))
-    roc_auc = engine.run(data, max_epochs=1).metrics["roc_auc"]
+        res = roc_auc.compute()
+        assert isinstance(res, float)
+        assert roc_auc_score(np_y, np_y_pred) == pytest.approx(res)
 
-    assert roc_auc == np_roc_auc
+    test_cases = [
+        (torch.randint(0, 2, size=(10,)).long(), torch.randint(0, 2, size=(10,)).long(), 1),
+        (torch.randint(0, 2, size=(100,)).long(), torch.randint(0, 2, size=(100,)).long(), 1),
+        (torch.randint(0, 2, size=(10, 1)).long(), torch.randint(0, 2, size=(10, 1)).long(), 1),
+        (torch.randint(0, 2, size=(100, 1)).long(), torch.randint(0, 2, size=(100, 1)).long(), 1),
+        # updated batches
+        (torch.randint(0, 2, size=(10,)).long(), torch.randint(0, 2, size=(10,)).long(), 16),
+        (torch.randint(0, 2, size=(100,)).long(), torch.randint(0, 2, size=(100,)).long(), 16),
+        (torch.randint(0, 2, size=(10, 1)).long(), torch.randint(0, 2, size=(10, 1)).long(), 16),
+        (torch.randint(0, 2, size=(100, 1)).long(), torch.randint(0, 2, size=(100, 1)).long(), 16),
+    ]
 
+    for y_pred, y, n_iters in test_cases:
+        # check multiple random inputs as random exact occurencies are rare
+        for _ in range(10):
+            _test(y_pred, y, n_iters)
 
-def test_integration_roc_auc_score_with_activated_output_transform():
 
-    np.random.seed(1)
-    size = 100
-    np_y_pred = np.random.rand(size, 1)
-    np_y_pred_sigmoid = torch.sigmoid(torch.from_numpy(np_y_pred)).numpy()
-    np_y = np.zeros((size,), dtype=np.long)
-    np_y[size // 2 :] = 1
-    np.random.shuffle(np_y)
+def test_multilabel_input_N():
 
-    np_roc_auc = roc_auc_score(np_y, np_y_pred_sigmoid)
+    roc_auc = ROC_AUC()
 
-    batch_size = 10
+    def _test(y_pred, y, n_iters):
```

Comment:
What does this test check, it does not call `compute` and there is not consistency checking ? Am I missing something ?